Makefile

```diff
@@ -26,6 +26,12 @@ aws-cloud-controller-manager: $(SOURCES)
 		-o=aws-cloud-controller-manager \
 		cmd/aws-cloud-controller-manager/main.go
 
+ecr-credential-provider:  $(shell find ./cmd/ecr-credential-provider -name '*.go')
+	 GO111MODULE=on CGO_ENABLED=0 GOOS=$(GOOS) GOPROXY=$(GOPROXY) go build \
+		-ldflags="-w -s -X 'main.version=$(VERSION)'" \
+		-o=ecr-credential-provider \
+		cmd/ecr-credential-provider/*.go
```

Comment:
I don't think the `*.go` is necessary, can just be `cmd/ecr-credential-provider` or `cmd/ecr-credential-provider/main.go`

Reply:
Unfortunately that doesn't work. I either need to build all go files together or move the framework into `/pkg`. Otherwise it complains about undefined `NewCredentialProvider`.

It's completely counter-intuitive to me, but based on my research go doesn't like having it in the `main` package.

Reply:
Hmm, I haven't seen the need to use `*.go`, what do you mean by 'doesn't work'?

Reply:
```
ip-172-31-62-224  cloud-provider-aws git:(ecr_creds_provider) ✗ 12/15 17:25 make ecr-credential-provider
GO111MODULE=on CGO_ENABLED=0 GOOS=linux GOPROXY=https://proxy.golang.org,direct go build \
        -ldflags="-w -s -X 'main.version=ec091a51-dirty'" \
        -o=ecr-credential-provider \
        cmd/ecr-credential-provider/main.go
# command-line-arguments
cmd/ecr-credential-provider/main.go:154:7: undefined: NewCredentialProvider
Makefile:30: recipe for target 'ecr-credential-provider' failed
make: *** [ecr-credential-provider] Error 2
```

Which is interesting because vscode can correctly locate it.

Reply:
Ok I think we can just fix this in a follow up PR if necessary, not a big deal